src/views/profile/bookmarks-view/helpers/render-bookmark-cards.tsx

```diff
@@ -0,0 +1,49 @@
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { formatTutorialData } from 'lib/learn-client/api/tutorial/formatting'
+import { ProductUsed } from 'lib/learn-client/types'
+import { ApiBookmark } from 'lib/learn-client/api/api-types'
+import TutorialCard from 'components/tutorial-card'
+import getReadableTime from 'components/tutorial-meta/components/badges/helpers'
+import { BookmarkButtonIconOnly } from 'components/bookmark-button'
+
+/**
+ * TODO: the bookmark button `handleClick` should
+ * render a confirmation dialog that can update data
+ */
+
+export default function renderBookmarkCard(bookmark: ApiBookmark) {
+	const {
+		id,
+		slug,
+		description,
+		readTime,
+		handsOnLab,
+		video,
+		name,
+		collectionCtx,
+		productsUsed,
+	} = formatTutorialData(bookmark.tutorial)
+	const duration = getReadableTime(readTime)
+	return (
+		<li key={id}>
+			<TutorialCard
+				heading={name}
+				description={description}
+				url={getTutorialSlug(slug, collectionCtx.default.slug)}
+				duration={duration}
+				hasInteractiveLab={Boolean(handsOnLab)}
+				hasVideo={Boolean(video)}
+				productsUsed={productsUsed.map((p: ProductUsed) => p.product.slug)}
+				eyebrowSlot={
+					<>
+						<span>{duration}</span>
+						<BookmarkButtonIconOnly
+							isBookmarked={true}
+							handleOnClick={() => console.log('Trigger confirmation dialog!')}
+						/>
+					</>
+				}
```

Comment:
I worry about this adding unnecessary maintenance complexity because it adds another piece of code where we'll need to make changes any time a tutorial card eyebrow changes. Can we leverage `TutorialCardWithBookmark` here by adding some props to it?

Reply:
This button interaction needs to behave much differently, it renders a different dialog and doesn't need to check for auth etc. So this seems like a better approach to me!

Reply:
Imo, a case like this is why we have the slot pattern. Because it needs slightly different treatment and this easily allows us to do that!

Reply:
Totally understand where you're coming from! The tradeoff here is that we have duplicate code for the same presentations, so if we have to make updates to the presentation of tutorial cards with bookmark buttons, we'll have to do it in two places. Like when Progress comes into play this week, won't it have to be implemented in two places with these changes? Here and in `TutorialCardWithBookmark`? Since the behavior of the button is the only thing that differs between the two, that's where I figure it would be useful to customize that with a prop passed to `TutorialCardWithBookmark`.

Reply:
I figure we can make abstractions for the progress work when we get there. I could see a future where we refactor all the eyebrows into one place when we're doing progress, which will make these different composed iterations easier to maintain together.

Reply:
💭  I dig this approach in the context of this PR - agreed we may want to abstract in the future (the "progress will have to go in two places" point seems like an excellent one); personally I don't feel we need to include that abstraction in this PR.